modules/collect/src/main/java/com/opengamma/collect/io/ResourceLocator.java

```diff
@@ -101,6 +107,29 @@ public static ResourceLocator ofClasspathUrl(URL url) {
     return new ResourceLocator(locator, Resources.asByteSource(url));
   }
 
+  /**
+   * Creates a stream of resource locators.
+   * <p>
+   * This finds all occurrences of the resource name in the classpath and returns them.
+   * 
+   * @param classpathResourceName  the classpath resource name
+   * @return the resource locators
+   */
+  @FromString
+  public static Stream<ResourceLocator> streamOfClasspathResources(String classpathResourceName) {
+    ArgChecker.notNull(classpathResourceName, "classpathResourceName");
+    try {
+      ClassLoader classLoader = firstNonNull(
+          Thread.currentThread().getContextClassLoader(),
```

Comment:
Based off principles used in Spring and elsewhere - use thread classloader and fallback if null